src/S3Commands.hh

```diff
@@ -260,7 +263,9 @@ class AmazonS3List : public AmazonRequest {
   public:
 	AmazonS3List(const S3AccessInfo &ai, const std::string &objectName,
 				 size_t maxKeys, XrdSysError &log)
-		: AmazonRequest(ai, objectName, log), m_maxKeys(maxKeys) {}
+		: AmazonRequest(ai, objectName, log, false), m_maxKeys(maxKeys) {
+		retain_object = false;
```

Comment:
This sets `retainObject` in two different ways.  Pick one.

If you keep the protected variable, do something like this instead:
```
		: AmazonRequest(ai, objectName, log, false), m_maxKeys(maxKeys), retainObject(false) {
```